src/index.js

```diff
@@ -54,226 +56,301 @@ function buildNametoIdMap(objectArr, type) {
-      const transformationName = transformationDict[transformerVersionID].name;
-      const transformationHandleName = _.camelCase(transformationName);
+    if (expectedOutput == "") {
+      continue;
+    }
 
-      fs.writeFileSync(
-        `${testOutputDir}/${transformationHandleName}_output.json`,
-        JSON.stringify(apiOutput, null, 2)
+    if (!isEqual(expectedOutput, apiOutput)) {
+      core.info(
+        `Test output do not match for transformation: ${transformationName}`
       );
-      testOutputFiles.push(
-        `${testOutputDir}/${transformationHandleName}_output.json`
+      outputMismatchResults.push(
+        `Test output do not match for transformation: ${transformationName}`
       );
 
-      if (
-        !transformationDict[transformerVersionID].hasOwnProperty(
-          "expected-output"
-        )
-      ) {
-        continue;
-      }
-
-      let expectedOutputfile =
-        transformationDict[transformerVersionID]["expected-output"];
-      let expectedOutput = expectedOutputfile
-        ? JSON.parse(fs.readFileSync(expectedOutputfile))
-        : "";
-
-      if (expectedOutput == "") {
-        continue;
-      }
-
-      if (!isEqual(expectedOutput, apiOutput)) {
-        core.info(
-          `Test output do not match for transformation: ${transformationName}`
-        );
-        outputMismatchResults.push(
-          `Test output do not match for transformation: ${transformationName}`
-        );
-
-        fs.writeFileSync(
-          `${testOutputDir}/${transformationHandleName}_diff.json`,
-          JSON.stringify(detailedDiff(expectedOutput, apiOutput), null, 2)
-        );
-
-        testOutputFiles.push(
-          `${testOutputDir}/${transformationHandleName}_diff.json`
-        );
-      }
-    }
+      fs.writeFileSync(
+        `${testOutputDir}/${transformationHandleName}_diff.json`,
+        JSON.stringify(detailedDiff(expectedOutput, apiOutput), null, 2)
+      );
 
-    // upload artifact
-    if (uploadTestArtifact === "true") {
-      core.info("Uploading test api output...");
-      await artifactClient.uploadArtifact(
-        "transformer-test-results",
-        testOutputFiles,
-        "."
+      testOutputFiles.push(
+        `${testOutputDir}/${transformationHandleName}_diff.json`
       );
     }
+  }
+}
+
+// Upload the test results to an artifact store.
+async function uploadTestArtifacts(testOutputFiles) {
+  // upload artifact
+  if (uploadTestArtifact) {
+    core.info("Uploading test api output");
+    await artifactClient.uploadArtifact(
+      "transformer-test-results",
+      testOutputFiles,
+      "."
+    );
+  }
+}
+
+// Publish the transformations and libraries.
+async function publishTransformation(
+  transformationTest,
+  librariesTest,
+  commitId
+) {
+  // publish
+  res = await publish(transformationTest, librariesTest, commitId);
+  core.info(`Publish result: ${JSON.stringify(res.data)}`);
```

Comment:
Please check if we have an incentive to print the outcome of publising to our customers. If not, please remove it ?

Reply:
Can you make sure, we need to print everything in the result of publish ?!

Reply:
Publish result: {"published":true}

We are getting this in the response.

Reply:
It's that's the only response, then does it make sense to log the response. Wouldn't it be implied that everything worked fine.